### DISCUSSION TITLE
Nodes and Edges Zindex Setting

### DISCUSSION THREAD

**Discussion by wilsoncwhuang** [xyflow/xyflow#4285] [Q&A]
Hi Community!

I recently built a topology visualization tool using react flow while I am struggling with setting z-index for nodes and edges. My purpose is to show the other block when the mouse enters the table cell as the snapshot below. I tried to set it in the style attribute of the tag or by the classname provided by react flow documentation but seems not working. How can I deal with this problem? Thank you.

```javascript
import { BaseEdge, getStraightPath } from 'reactflow';
import { tokens } from "../theme.js";
import { useTheme } from "@mui/material";

function DetailStraightEdge({ id, sourceX, sourceY, targetX, targetY, sourcePosition, targetPosition, data }) {
    
    const theme = useTheme();
    const colors = tokens(theme.palette.mode);

    const [edgePath, labelX, labelY] = getStraightPath({
        sourceX,
        sourceY,
        sourcePosition,
        targetPosition,
        targetX,
        targetY,
    });

    return (
        <>
            <BaseEdge path={edgePath} style={{strokeWidth:2, stroke:colors.grey[200], zIndex:-100}}/>
        </>
    )
}
export default DetailStraightEdge;
``` 

![image](https://github.com/xyflow/xyflow/assets/98676453/a928d40f-8962-4f9a-842c-9b388d687d04)

**Comment by moklick** [comment]
React Flow renders an edge in a different wrapper if you use the `edge.zIndex` attribute. You can add only adjust the z-index by setting that attribute.
